pkgs/by-name/pr/prometheus/package.nix

```
@@ -55,48 +56,71 @@ buildGoModule (finalAttrs: {
     hash = "sha256-oOEvNZFlYtTNBsn+B2pAWXi0A2oJ6IAo7V8bOLtjfCM=";
   };
 
+  buildHostGenerated = buildPackages.buildGoModule {
```

```suggestion
  # Go started using the compiler even with `CGO_ENABLED=0`. Therefore, either this separate
  # `buildGoModule` approach must be used, or the approach taken in 756865d780f3c4403cb9dc59a08303561bfc26e7.
  # `aarch64-unknown-linux-gnu-gcc: error: unrecognized command-line option '-m64'`
  buildHostGenerated = buildPackages.buildGoModule {
```

https://github.com/NixOS/nixpkgs/pull/465092/commits/756865d780f3c4403cb9dc59a08303561bfc26e7

---

I actually like this solution much more. It's also much more self-evident what it does.